src/descriptors.rs

```diff
@@ -108,6 +133,21 @@ impl<'a> FieldDescriptor<'a> {
     }
 }
 
+impl fmt::Display for FieldDescriptor<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        if self.dimensions > 0 {
```

Comment:
Don't need the if/else here, if dimensions is 0 then the repeat call will already return an empty string. So the if branch already handles both cases correctly